helm: add support to pass --namespace option

Yuggupta27 opened the pull request:
# Describe what this PR does #

The current approach uses hard-coded command-line arguments which are not very robust as it just depends
on positional arguments;

We would like to allow the possibility to add some additional flags as arguments (deploy_sc and deploy_secret 
to deploy storageclass and secrets) 
See: #1786

The reason why we are switching from `./scripts/install-helm.sh install-ceph-csi ${PASSED_NAMESPACE} ` to `./scripts/install-helm.sh install-ceph-csi --namespace ${PASSED_NAMESPACE}`  is because, In order to parse the arguments and identify known options, it is very difficult to do if we have an unknown string in the arguments(namespace).

To maintain backward compatibility, the script will also keep working as the previous approach(`./scripts/install-helm.sh install-ceph-csi ${PASSED_NAMESPACE}` ) which can be deprecated in the coming future.

Updates: #1786 #2013 
Signed-off-by: Yug <yuggupta27@gmail.com>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

Yuggupta27 committed:
helm: add support to pass --namespace option

The current approach uses hard-coded command line
arguments which is not very robust;
To maintain backward compatibility, script will
also keep working as the previous approach.

Signed-off-by: Yug <yuggupta27@gmail.com>

Yuggupta27 commented:
@nixpanic @Madhu-1 as discussed earlier, this PR will help us unblock both #1786 and #2013

Yuggupta27 commented:
/retest ci/centos/k8s-e2e-external-storage/1.20

humblec commented:
> /retest ci/centos/k8s-e2e-external-storage/1.20

@Yuggupta27  #2196 ?

Madhu-1 commented:
> > /retest ci/centos/k8s-e2e-external-storage/1.20
> 

external-test results are not mandatory to merge the PR but its good to improve it. Please Don't retest it unless your PR is fixing it.

Yuggupta27 commented:
> > /retest ci/centos/k8s-e2e-external-storage/1.20
> 
> @Yuggupta27 #2196 ?

Yes, It's the same issue.

nixpanic commented on scripts/install-helm.sh:
```diff
@@ -21,6 +21,27 @@ NODE_LABEL_ZONE="test.failure-domain/zone"
 REGION_VALUE="testregion"
 ZONE_VALUE="testzone"
 
+example() {
```
other functions use the `function` keyword. keep coding style the same for this file please

Yuggupta27 replied:
The majority of functions in this script don't make use of the `function` keyword. Should I get rid of the `function` keyword since only 2 functions make use of it to make it uniform?

Yuggupta27 replied:
Added a commit to make sure the convention is the same throughout the file.

nixpanic commented on scripts/install-helm.sh:
```diff
@@ -163,24 +184,72 @@ if ! helm_loc="$(type -p "helm")" || [[ -z ${helm_loc} ]]; then
+fi
+
+if [ ${#SKIP_PARSE} -eq 0 ]; then
+    while [ "$1" != "" ];
```
the `;` is not needed, unless you place `do` on the same line

nixpanic commented on scripts/install-helm.sh:
```diff
@@ -163,24 +184,72 @@ if ! helm_loc="$(type -p "helm")" || [[ -z ${helm_loc} ]]; then
     ;;
 install-cephcsi)
-    install_cephcsi_helm_charts "$2"
+    install_cephcsi_helm_charts "$NAMESPACE"
```
use `${NAMESPACE}` notation please

nixpanic reviewed (changes requested):
just some coding style improvements needed

Yuggupta27 committed:
helm: remove function keyword

Getting rid of function keyword for two reasons:
1. Defining a function without 'function' keyword is more
   portable as it is compatible with Bourne/Korn/POSIX scripts
2. To ensure the coding style is same for the file.

Signed-off-by: Yug <yuggupta27@gmail.com>

Yuggupta27 commented:
/retest ci/centos/mini-e2e/k8s-1.19

Yuggupta27 commented:
/retest ci/centos/mini-e2e/k8s-1.19

Yuggupta27 commented:
/retest ci/centos/upgrade-tests-cephfs

Yuggupta27 commented:
> /retest ci/centos/upgrade-tests-cephfs

Rook deployment failed
